layouts/applications/list.html

```diff
@@ -112,4 +122,133 @@ <h5 class="card-title d-flex align-items-center">
+        <input type="checkbox" value="${prefix}-${item}" class="custom-control-input" id="${prefix}-${item}" aria-label="${prefix}-${item}">
+        <label class="custom-control-label" for="${prefix}-${item}">${lookup[item] || item}</label>
+      </div>
+      `;
+    };
+    var buildFiltersUI = function () {
+      let services = "",
+        deployments = "",
+        platforms = "";
+
+      $items.each(function (_i, value) {
+        let service = value.getAttribute("data-services"),
+          platform = value.getAttribute("data-platforms"),
+          deployment = value.getAttribute("data-deployments");
+
+        services += service ? service.split(",") + "," : "";
+        platforms += platform ? platform.split(",") + "," : "";
+        deployments += deployment ? deployment.split(",") + "," : "";
+      });
+
+      // Set removes duplicates
+      services = Array.from(new Set(services.slice(0, -1).split(",")));
+      platforms = Array.from(new Set(platforms.slice(0, -1).split(",")));
+      deployments = Array.from(new Set(deployments.slice(0, -1).split(",")));
+
+      let servicesHtml = "",
+        deploymentsHtml = "",
+        platformsHtml = "";
+
+      services.sort().map(function (service) {
+        servicesHtml += checkboxItemHtmlGenerator("service", servicesData, service);
+      });
+      deployments.sort().map(function (deployment) {
+        deploymentsHtml += checkboxItemHtmlGenerator("deployment", deploymentsData, deployment);
+      });
+      platforms.sort().map(function (platform) {
+        platformsHtml += checkboxItemHtmlGenerator("platform", platformsData, platform);
+      });
+
+      $("#filters-services-container").html(servicesHtml);
+      $("#filter-deployments-container").html(deploymentsHtml);
+      $("#filter-platforms-container").html(platformsHtml);
+    };
+
+    var bindFilterEvents = function () {
+      var $filterCheckboxes = $("#sidebar-filters-parent .custom-control-input");
+
+      $clearFiltersBtn.on("click", function () {
+        filtersState = Object.keys(filtersState).reduce(function (acc, item) {
+          return Object.assign(acc, {[item]: []})
+        }, {})
+        $('aside').find($clearFiltersBtn).hide();
+        $filterCheckboxes.prop("checked", false);
+        updateApplicationsUI();
+      });
+      $filterCheckboxes.on("change", function (e) {
+        let type = `${e.target.value.split("-")[0]}s`;
+        let arr = e.target.checked
+          ? filtersState[type].concat(e.target.value)
+          : filtersState[type].filter(function (item) {
+              return item !== e.target.value;
+            });
+        Object.assign(filtersState, { [type]: arr });
+        updateApplicationsUI();
+      });
+    };
+
+    var isFiltersApplied = function () {
+      let applied = false;
+      Object.keys(filtersState).forEach(function (key) {
+        if (filtersState[key].length !== 0) applied = true;
+      });
+      return applied;
```

Comment:
seems like we could improve this by early exiting the function on line 210 with `return true` and replace line 212 with `return false`.

Reply:
I did the exact same at first but unfortunately `.forEach` [can't be skipped from in between](https://stackoverflow.com/a/2641374/10228249) and I don't usually use for loops to iterate arrays so I went ahead with this solution. One option would be to use `.some` / `.every` to resolve it if you think this is an issue

Reply:
interesting, today I learned!